presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -85,14 +85,12 @@ public DictionaryBlock(int positionCount, Block dictionary, Slice ids, boolean d
         else {
             int sizeInBytes = 0;
             int uniqueIds = 0;
-            boolean[] isReferenced = getReferencedPositions(dictionary, ids, positionCount);
-            for (int position = 0; position < isReferenced.length; position++) {
-                if (isReferenced[position]) {
-                    if (!dictionary.isNull(position)) {
-                        sizeInBytes += dictionary.getLength(position);
-                    }
-                    uniqueIds++;
+            for (int i = 0; i < positionCount; i++) {
+                int position = getIndex(ids, i);
+                if (!dictionary.isNull(position)) {
+                    sizeInBytes += dictionary.getLength(position);
                 }
+                uniqueIds++;
```

Comment:
This change doesn't seem correct or at least changes the definition of sizeInBytes and uniqueIds.  The new code is counting the all positions even if they refer to the same dictionary entry, so anything that is relying on these numbers will need to be adjusted.

@nileema should review this